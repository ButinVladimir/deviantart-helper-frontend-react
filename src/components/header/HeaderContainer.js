import { connect } from 'react-redux';
import Header from './Header';

/**
 * @description
 * Maps Redux state to props.
 *
 * @param {State} state - The state.
 * @returns {Object} Props.
 */
const mapStateToProps = state => ({
  userState: state.shared.userState,
  menuToggled: state.shared.menuToggled,
});

export default connect(mapStateToProps)(Header);
