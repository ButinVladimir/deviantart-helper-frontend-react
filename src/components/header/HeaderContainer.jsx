import { connect } from 'react-redux';
import Header from './Header';
import userRevokeActionCreator from '../../redux/action-creators/user/revoke';
import userRefreshActionCreator from '../../redux/action-creators/user/refresh';
import deviationsLoadActionCreator from '../../redux/action-creators/deviations/load';

/**
 * @description
 * Maps Redux state to header props.
 *
 * @param {State} state - The state.
 * @returns {Object} Props.
 */
const mapStateToProps = state => ({
  isLoggedIn: state.shared.isLoggedIn,
  userIcon: state.user.userIcon,
});

/**
 * @description
 * Maps Redux dispatch to header props.
 *
 * @param {Function} dispatch - The dispatch.
 * @param {Object} ownProps - Component own props.
 * @returns {Object} Props.
 */
const mapDispatchToProps = (dispatch, ownProps) => ({
  revokeHandler: () => dispatch(userRevokeActionCreator(ownProps.config)),
  refreshHandler: () => dispatch(userRefreshActionCreator(ownProps.config)),
  deviationsLoadHandler: () => dispatch(deviationsLoadActionCreator(ownProps.config)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
