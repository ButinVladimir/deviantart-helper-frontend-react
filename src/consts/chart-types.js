export const CHART_VIEWS = 'views';
export const CHART_FAVOURITES = 'favourites';
export const CHART_COMMENTS = 'comments';
export const CHART_DOWNLOADS = 'downloads';
