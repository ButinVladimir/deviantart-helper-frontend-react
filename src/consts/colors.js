const LIGHT_BLUE = '#88f';
const RED = '#b33';
const GREEN = '#393';
const BLUE = '#11c';
const ORANGE = '#f94';
const PURPLE = '#94a';
const CYAN = '#2b9';
const BLACK = '#111';
const GREY = '#999';
const PINK = '#d5b';

export default [LIGHT_BLUE, RED, GREEN, BLUE, ORANGE, PURPLE, CYAN, BLACK, GREY, PINK];
