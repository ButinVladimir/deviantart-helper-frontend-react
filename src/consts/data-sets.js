export const DATA_SET_TIMESTAMP = 0;
export const DATA_SET_VIEWS = 1;
export const DATA_SET_FAVOURITES = 2;
export const DATA_SET_COMMENTS = 3;
export const DATA_SET_DOWNLOADS = 4;

export const dataSetOptions = [
  [DATA_SET_VIEWS, 'Views'],
  [DATA_SET_FAVOURITES, 'Favourites'],
  [DATA_SET_COMMENTS, 'Comments'],
  [DATA_SET_DOWNLOADS, 'Downloads'],
];
