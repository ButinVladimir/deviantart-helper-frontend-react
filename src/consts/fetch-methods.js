export const GET = 'GET';
export const POST = 'POST';
