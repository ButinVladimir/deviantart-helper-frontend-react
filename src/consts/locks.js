export const LOCK_LOAD_USER_INFO = 'LOCK_LOAD_USER_INFO';
export const LOCK_START_LOADING_DATA = 'LOCK_START_LOADING_DATA';
export const LOCK_REVOKE = 'LOCK_REVOKE';

export const LOCK_BROWSE_DEVIATIONS = 'LOCK_BROWSE_DEVIATIONS';
export const LOCK_DEVIATION_DETAILS = 'LOCK_DEVIATION_DETAILS';
export const LOCK_DEVIATION_DETAILS_METADATA = 'LOCK_DEVIATION_DETAILS_METADATA';
export const LOCK_DEVIATIONS_STATISTICS = 'LOCK_DEVIATIONS_STATISTICS';
export const LOCK_DEVIATIONS_TOTAL = 'LOCK_DEVIATIONS_TOTAL';
