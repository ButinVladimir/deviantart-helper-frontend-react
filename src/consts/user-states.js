export const NOT_LOGGINED = 0;
export const FETCHING_INFO = 1;
export const FULLY_LOGINNED = 2;
