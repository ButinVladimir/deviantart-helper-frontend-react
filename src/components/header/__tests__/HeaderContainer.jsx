import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import HeaderContainer from '../HeaderContainer';
import createDefaultState from '../../../redux/states/state';
import * as userStates from '../../../consts/user-states';

jest.mock('../../../redux/action-creators/shared/hide-message', () => jest.fn(() => ({ type: 'TEST_ACTION' })));

describe('HeaderContainer', () => {
  it('renders correctly', () => {
    const state = {
      ...createDefaultState(),
      shared: {
        userState: userStates.FETCHING_INFO,
        menuToggled: true,
        revokeLoading: false,
      },
    };

    const mockStore = configureStore([thunk]);
    const store = mockStore(state);

    const wrapper = mount(
      <Provider store={store}>
        <HeaderContainer />
      </Provider>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
