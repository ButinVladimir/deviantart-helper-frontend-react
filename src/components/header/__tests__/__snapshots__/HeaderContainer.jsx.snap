// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer can be rendered correctly 1`] = `
<nav
  className="navbar is-dark"
  role="navigation"
  style={Object {}}
>
  <div
    className="container is-fullhd"
    style={Object {}}
  >
    <div
      className="navbar-menu is-active"
      style={Object {}}
    >
      <div
        className="navbar-end"
        style={Object {}}
      >
        <a
          className="navbar-item"
          style={Object {}}
        >
          <div
            className="buttons"
          >
            <button
              className="is-danger button"
              disabled={false}
              id="revoke-button"
              onClick={[Function]}
              style={Object {}}
              tabIndex={0}
            >
              Log out
            </button>
          </div>
        </a>
      </div>
    </div>
  </div>
</nav>
`;
